sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -115,6 +157,8 @@ object TypeEnum extends Enumeration {
   val MAP: Value = Value
   val STRUCT: Value = Value
   val UDT: Value = Value
+  val DAYTIME: Value = Value
+  val YEARMONTH: Value = Value
 }
```

Comment:
Can we util ShimLoader.getSparkVersion to construct static version to set(TypeEnum) map?
Not know if feasible, if ok, just not need the spark 320 code.

Reply:
Adding more enums values dynamically is not something that scala supports as far as I can tell.